### PR TITLE
Implemented #14

### PR DESCRIPTION
**Pull request by Asdamp**
implemented an option to refuse artifacts that has the same version of an artifact already uploaded

### CODE REVIEW

**Comment by renaudcerrato** [comment]
Sorry for being late, just overbusy those days to do a review. Will do asap, thanks for your patience :)

**Comment by Asdamp** [comment]
No problem. take your time

**Comment by Asdamp** [comment]
Any news on this?

**Comment by renaudcerrato** [comment]
Hi there! I totally forgot this one 😅 Will do the review once I'll be at the office.

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+uniqueartifacts=true
```
Comment (renaudcerrato) [inline src/main/resources/config.properties]: Could you instead use system properties from appengine-web.xml?

File: src/main/java/repo/resource/RepositoryResource.java [hunk 2]
```diff
@@ -149,7 +154,35 @@ public Response put(@PathParam("file") String file,
...
+
+    private boolean isNotAMavenFile(String file) {
+        return !file.endsWith("maven-metadata.xml") && !file.endsWith("maven-metadata.xml.sha1") && !file.endsWith("maven-metadata.xml.md5");
+    }
+
+    private boolean gcsFileExist(GcsFilename filename) throws IOException {
+        return gcs.getMetadata(filename)!=null;
+    }
+
+    private boolean isuniqueEnabled(){
```
Comment (renaudcerrato) [inline src/main/java/repo/resource/RepositoryResource.java]: Overkill, please use system properties (appengine-web.xml)

File: src/main/java/repo/resource/RepositoryResource.java [hunk 3]
```diff
@@ -149,7 +154,35 @@ public Response put(@PathParam("file") String file,
             options.mimeType(mimeType);
         }
 
+
+        if(isuniqueEnabled() && gcsFileExist(filename) && isNotAMavenFile(file)){
+            LOGGER.info("The uploaded artifact is already inside the repository. If you want to overwrite the artifact, you have to disable the 'uniqueartifacts' flag");
+            return Response.notAcceptable(null).entity("The uploaded artifact is already inside the repository. If you want to overwrite the artifact, you have to disable the 'uniqueartifacts' flag").build();
```
Comment (renaudcerrato) [inline src/main/java/repo/resource/RepositoryResource.java]: Duplicated string , could you refactor?

**Pull request closed without merging**